### PR TITLE
Add parser for QSFP-DD cable type and dictionaries for QSFP-DD codes

### PR DESCRIPTION
**Pull request by shlomibitton**
**- What I did**
Add parser for QSFP-DD cable type.
Add dictionaries for QSFP-DD codes.

**- How I did it**
**- How to verify it**
Check for Transceiver DB data with QSFP-DD cable.
Check 'show interfaces transceiver eeprom Ethernet#' output.

**- Previous command output (if the output of a command-line utility has changed)**
**- New command output (if the output of a command-line utility has changed)**

### CODE REVIEW

File: sonic_platform_base/sonic_sfp/sff8024.py [hunk 1]
```diff
@@ -64,3 +64,238 @@
...
+	'00': 'Undefined',
+    '01': 'Active Cable assembly with BER < 10^-12',
+    '02': 'Active Cable assembly with BER < 5x10^-5',
+    '03': 'Active Cable assembly with BER < 2.6x10^-4',
+    '04': 'Active Cable assembly with BER < 10^-6',
+    'bf': 'Active Loopback module',
+    }
+
+base_t_media_interface = {
+	'00': 'Undefined',
```
Comment (keboliu) [inline sonic_platform_base/sonic_sfp/sff8024.py:296]: align indent

File: sonic_platform_base/sonic_sfp/sff8024.py [hunk 2]
```diff
@@ -64,3 +64,238 @@
...
+    '3a': '25G-SR',
+    '3b': '25G-LR',
+    '3c': '10G-LR-BiDi',
+    '3d': '25G-LR-BiDi',
+    '3e': '400ZR, DWDM, amplified',
+    '3f': '400ZR, Single Wavelength, Unamplified',
+    }
+
+passive_copper_media_interface = {
+	'00': 'Undefined',
```
Comment (keboliu) [inline sonic_platform_base/sonic_sfp/sff8024.py:281]: align indent

File: sonic_platform_base/sonic_sfp/sff8024.py [hunk 3]
```diff
@@ -64,3 +64,238 @@
...
+    '2c': '4I1-9D1F (G.959.1)',
+    '2d': '4L1-9C1F (G.959.1)',
+    '2e': '4L1-9D1F (G.959.1)',
+    '2f': 'C4S1-9D1F (G.695)',
+    '30': 'C4S1-4D1F (G.695)',
+    '31': '4I1-4D1F (G.959.1)',
+    '32': '8R1-4D1F (G.959.1)',
+    '33': '8I1-4D1F (G.959.1)',
+    '38': '10G-SR',
+	'39': '10G-LR',
```
Comment (keboliu) [inline sonic_platform_base/sonic_sfp/sff8024.py:271]: align indent

File: sonic_platform_base/sonic_sfp/sff8024.py [hunk 4]
```diff
@@ -64,3 +64,238 @@
...
+    '10': '100G CWDM4 MSA Spec',
+    '11': '100G 4WDM-10 MSA Spec',
+    '12': '100G 4WDM-20 MSA Spec',
+    '13': '100G 4WDM-40 MSA Spec',
+    '14': '100GBASE-DR (Cl 140)',
+    '15': '100G-FR/100GBASE-FR1 (Cl 140)',
+    '16': '100G-LR/100GBASE-LR1 (Cl 140)',
+    '17': '200GBASE-DR4 (Cl 121)',
+    '18': '200GBASE-FR4 (Cl 122)',
+	'19': '200GBASE-LR4 (Cl 122)',
```
Comment (keboliu) [inline sonic_platform_base/sonic_sfp/sff8024.py:244]: align indent

File: sonic_platform_base/sonic_sfp/sff8024.py [hunk 5]
```diff
@@ -64,3 +64,238 @@
...
+    '14': '10GFC-MM (10GFC)',
+    '15': '16GFC-MM (FC-PI-5)',
+    '16': '32GFC-MM (FC-PI-6)',
+    '17': '64GFC-MM (FC-PI 7)',
+    '18': '128GFC-MM4 (FC-PI-6P)',
+    '19': '256GFC-MM4 (FC-PI-7P)',
+    }
+
+sm_media_interface = {
+	'00': 'Undefined',
```
Comment (keboliu) [inline sonic_platform_base/sonic_sfp/sff8024.py:217]: align indent

File: sonic_platform_base/sonic_sfp/sff8024.py [hunk 6]
```diff
@@ -64,3 +64,238 @@
...
+    '3b': 'FOIC1.4 (ITU-T G.709.1/Y.1331 G.Sup58)',
+    '3c': 'FOIC1.2 (ITU-T G.709.1/Y.1331 G.Sup58)',
+    '3d': 'FOIC2.8 (ITU-T G.709.1/Y.1331 G.Sup58)',
+    '3e': 'FOIC2.4 (ITU-T G.709.1/Y.1331 G.Sup58)',
+    '3f': 'FOIC4.16 (ITU-T G.709.1 G.Sup58)',
+    '40': 'FOIC4.8 (ITU-T G.709.1 G.Sup58)', 
+    }
+
+nm_850_media_interface = {
+	'00': 'Undefined',
```
Comment (keboliu) [inline sonic_platform_base/sonic_sfp/sff8024.py:187]: align indent

File: sonic_platform_base/sonic_sfp/qsfp_dd.py [hunk 7]
```diff
@@ -0,0 +1,753 @@
...
+        }
+
+    vendor_oui = {
+        'Vendor OUI':
+                {'offset':0,
+                 'size':3,
+                 'type' : 'hex'}
+        }
+
+    qsfp_dom_capability = {
```
Comment (keboliu) [inline sonic_platform_base/sonic_sfp/qsfp_dd.py:180]: qsfp --> qsfp-dd ?

File: sonic_platform_base/sonic_sfp/qsfp_dd.py [hunk 8]
```diff
@@ -0,0 +1,753 @@
...
+    def parse_media_type(self, qsfp_media_type_raw_data, start_pos):
+        return self.decode_media_type(qsfp_media_type_raw_data, start_pos, 1)
+
+    def dump_pretty(self):
+        if self.interface_data is None:
+            print('Object not initialized, nothing to print')
+            return
+        sffbase.dump_pretty(self, self.interface_data)
+
+    def get_data(self):
```
Comment (keboliu) [inline sonic_platform_base/sonic_sfp/qsfp_dd.py:239]: what's this function for?

File: sonic_platform_base/sonic_sfp/qsfp_dd.py [hunk 9]
```diff
@@ -0,0 +1,753 @@
...
+
+    def get_data_pretty(self):
+        return sffbase.get_data_pretty(self, self.interface_data)
+
+
+class qsfp_dd_Dom(sffbase):
+
+    version = '1.0'
+
+    def calc_temperature(self, eeprom_data, offset, size):
```
Comment (keboliu) [inline sonic_platform_base/sonic_sfp/qsfp_dd.py:250]: Will the calibration type matter for temp/voltage/power... calculation?

File: sonic_platform_base/sonic_sfp/sff8024.py [hunk 10]
```diff
@@ -64,3 +64,238 @@
...
+    'bf': 'Active Loopback module',
+    }
+
+base_t_media_interface = {
+	'00': 'Undefined',
+    '01': '1000BASE-T (Clause 40)',
+    '02': '2.5GBASE-T (Clause 126)',
+    '03': '5GBASE-T (Clause 126)',
+    '04': '10GBASE-T (Clause 55)',
+    }
```
Comment (keboliu) [inline sonic_platform_base/sonic_sfp/sff8024.py:301]: new line?

File: sonic_platform_base/sonic_sfp/sff8024.py [hunk 11]
```diff
@@ -64,3 +64,238 @@
...
+
+passive_copper_media_interface = {
+	'00': 'Undefined',
+    '01': 'Copper cable',
+    '02': 'Passive Loopback module',
+    }
+
+active_cable_media_interface = {
+	'00': 'Undefined',
```
Comment (keboliu) [inline sonic_platform_base/sonic_sfp/sff8024.py:287]: align indent

File: sonic_platform_base/sonic_sfp/sff8024.py (hunk 1 above)
Comment (shlomibitton) [reply sonic_platform_base/sonic_sfp/sff8024.py]: Fixed.

File: sonic_platform_base/sonic_sfp/sff8024.py (hunk 2 above)
Comment (shlomibitton) [reply sonic_platform_base/sonic_sfp/sff8024.py]: Fixed.

File: sonic_platform_base/sonic_sfp/sff8024.py (hunk 3 above)
Comment (shlomibitton) [reply sonic_platform_base/sonic_sfp/sff8024.py]: Fixed.

File: sonic_platform_base/sonic_sfp/sff8024.py (hunk 4 above)
Comment (shlomibitton) [reply sonic_platform_base/sonic_sfp/sff8024.py]: Fixed.

File: sonic_platform_base/sonic_sfp/sff8024.py (hunk 5 above)
Comment (shlomibitton) [reply sonic_platform_base/sonic_sfp/sff8024.py]: Fixed.

File: sonic_platform_base/sonic_sfp/qsfp_dd.py (hunk 7 above)
Comment (shlomibitton) [reply sonic_platform_base/sonic_sfp/qsfp_dd.py]: Changed.

File: sonic_platform_base/sonic_sfp/qsfp_dd.py (hunk 8 above)
Comment (shlomibitton) [reply sonic_platform_base/sonic_sfp/qsfp_dd.py]: We don't need it, deleted it.

File: sonic_platform_base/sonic_sfp/qsfp_dd.py (hunk 9 above)
Comment (shlomibitton) [reply sonic_platform_base/sonic_sfp/qsfp_dd.py:237]: Arch team didn't say anything about calibration when I asked.
As for copper and optic cables, seems no calibration needed.

File: sonic_platform_base/sonic_sfp/sff8024.py (hunk 11 above)
Comment (shlomibitton) [reply sonic_platform_base/sonic_sfp/sff8024.py]: Fixed.

File: sonic_platform_base/sonic_sfp/sff8024.py (hunk 10 above)
Comment (shlomibitton) [reply sonic_platform_base/sonic_sfp/sff8024.py]: Added.

File: sonic_platform_base/sonic_sfp/qsfp_dd.py [hunk 12]
```diff
@@ -0,0 +1,729 @@
...
+        }
+
+    vendor_date = {
+        'VendorDataCode(YYYY-MM-DD Lot)':
+                {'offset':0,
+                'size':8,
+                'type': 'date'}
+        }
+
+    def parse_sfp_type(self, type_raw_data, start_pos):
```
Comment (keboliu) [inline sonic_platform_base/sonic_sfp/qsfp_dd.py:194]: do we need a function  "parse_sfp_info_bulk" like  https://github.com/Azure/sonic-platform-common/blob/master/sonic_platform_base/sonic_sfp/sff8436.py#L485 ?

Comment (shlomibitton) [reply sonic_platform_base/sonic_sfp/qsfp_dd.py:194]: No, I parsed each field in a different function to make it more clear.

File: sonic_platform_base/sonic_sfp/sff8024.py (hunk 6 above)
Comment (shlomibitton) [reply sonic_platform_base/sonic_sfp/sff8024.py]: Fixed.

**Pull request closed without merging**